File: 04. CLICON/ClienteConsola/src/main/java/ec/edu/monster/clienteconsola/vista/OpcionMenu.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.clienteconsola.vista;

/**
 *
 * @author joanc
 */
public enum OpcionMenu {
        CONVERTIR_LONGITUD("1", "Convertir Longitud"),
        CONVERTIR_MASA("2", "Convertir Masa"),
        CONVERTIR_TEMPERATURA("3", "Convertir Temperatura"),
        SALIR("4", "Salir");

        private final String numero;
        private final String descripcion;

        OpcionMenu(String numero, String descripcion) {
            this.numero = numero;
            this.descripcion = descripcion;
        }

        public String getNumero() { return numero; }
        public String getDescripcion() { return descripcion; }
    }

File: SOAP_DOTNET/04. CLICON/Vista/OpcionMenu.cs
namespace ClienteConsola.Vista
{
    public sealed class OpcionMenu
    {
        public static readonly OpcionMenu CONVERTIR_LONGITUD = new OpcionMenu("1", "Convertir Longitud");
        public static readonly OpcionMenu CONVERTIR_MASA = new OpcionMenu("2", "Convertir Masa");
        public static readonly OpcionMenu CONVERTIR_TEMPERATURA = new OpcionMenu("3", "Convertir Temperatura");
        public static readonly OpcionMenu VALIDAR_NUMERICO = new OpcionMenu("4", "Validar Campo Numerico");
        public static readonly OpcionMenu SALIR = new OpcionMenu("5", "Salir");

        private readonly string numero;
        private readonly string descripcion;

        private OpcionMenu(string numero, string descripcion)
        {
            this.numero = numero;
            this.descripcion = descripcion;
        }

        public string getNumero() => numero;
        public string getDescripcion() => descripcion;

        public static OpcionMenu[] values() => new[]
        {
            CONVERTIR_LONGITUD,
            CONVERTIR_MASA,
            CONVERTIR_TEMPERATURA,
            VALIDAR_NUMERICO,
            SALIR
        };
    }
}
